lib/puppet/provider/foreman_smartproxy/rest_v3.rb

```diff
@@ -3,8 +3,19 @@
 
   def proxy
     @proxy ||= begin
-      r = request(:get, 'api/v2/smart_proxies', :search => %{name="#{resource[:name]}"})
-      raise Puppet::Error.new("Proxy #{resource[:name]} cannot be retrieved: #{error_message(r)}") unless success?(r)
+      path = 'api/v2/smart_proxies'
+      r = request(:get, path, :search => %{name="#{resource[:name]}"})
+
+      unless success?(r)
+        error_string = <<~ERROR.chomp
+
```

Comment:
Why this empty line?

Reply:
The effect is to split this into two lines, which I personally found to be more readable:

```
    2021-01-15 21:17:13 [ERROR ] [configure] /Stage[main]/Foreman_proxy::Register/Foreman_smartproxy[centos7-foreman-proxy-nightly.mercury.example.com]: Could not evaluate:
    2021-01-15 21:17:13 [ERROR ] [configure] SmartProxy centos7-foreman-proxy-nightly.mercury.example.com cannot be retrieved from Foreman:
```

Reply:
I've converted the output back to a single line